Update headercard to use filter to add ExPlat parameter

ilyasfoo opened the pull request:
Fixes #8227

This PR:
1. Adds the capability for tasklist progression experiments to append WCA's install timestamp to ExPlat API request
2. Changes the way experiment is called. Instead of using `useExperiment`, we're using a custom hook since we need to conditionally load the experiment after the required option is loaded

After a while of testing, I observed a noticeable delay in my local for the experiment tasklist to be displayed since it's not preloaded. We may need to consider preloading `woocommerce_admin_install_timestamp` option and hydrate if this is proven to be a problem.

### Detailed test instructions:

1. Since we're using dynamic experiments name, this needs to be tested anywhen in February 2022. Alternatively, hardcode the month [here](https://github.com/woocommerce/woocommerce-admin/blob/6824e721c21d01f9f635fe0c9359f0046fe4ba13/client/homescreen/hooks/use-headercard-experiment-hook.js#L46) to `'02'`
1. Use the treatment bookmarklet for experiment `woocommerce_tasklist_progression_headercard_2col_2022_02`
2. Open your browser's console network tab
3. Go to WooCommerce > Home
4. Observe the network request for `https://public-api.wordpress.com/wpcom/v2/experiments/0.1.0/assignments/woocommerce?experiment_name=woocommerce_tasklist_progression_headercard_2col_2022_02` contains the payload `woo_wcadmin_install_timestamp` and the value is a timestamp
5. If you received treatment variation, you should see the two column tasklist experiment

Commit:
Add changelog

ilyasfoo committed:
Update headercard experiment hook to utilize `woocommerce_explat_request_args` filter

ilyasfoo committed:
Add test

ilyasfoo committed:
Fix experiment name check

ilyasfoo committed:
Update parameter name

ilyasfoo committed:
Add changelog

louwie17 reviewed (changes requested):
> After a while of testing, I observed a noticeable delay in my local for the experiment tasklist to be displayed since it's not preloaded. We may need to consider preloading woocommerce_admin_install_timestamp option and hydrate if this is proven to be a problem.

I encountered the same thing, probably because I have XDebug enabled locally, so things are slower in general. But this causes some real jankiness, with showing the normal task list first:
![task-list-experiment-timestamp](https://user-images.githubusercontent.com/2240960/152003499-bb07cc3b-ddbf-4f76-8cc8-728ad869128f.gif)

I think pre-loading or making it available through `wcSettings.admin` would be the best bet. I would lean towards the latter.

ilyasfoo committed:
Removed old changelog

ilyasfoo committed:
Added prop comments to Layout component

ilyasfoo committed:
Add preload option for homescreen

ilyasfoo commented:
Thanks for reviewing this, @louwie17! I agree, I've used the preload option for Homescreen.php [015e0b9](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/015e0b907fbee993dc09b46573df8b0241d843c0) since it's already available.

Please have another look!

adrianduffell reviewed (commented):
This tested well for me! I'll leave it up to @louwie17 to have final review.

louwie17 reviewed (changes requested):
Thanks for the update @ilyasfoo, things are feeling better now, although there is still a bit of a flash between rendering the placeholder of the old task list and then the new task list.
I can't put my finger on what code exactly is causing this, but this seems to be the flow:
- Loading icon happens until the experiment is loaded
- Experiment is loaded and the inbox panel placeholder shows quickly with the Stats overview panel
- Then it loads the new experiment task list
- Clearing the UI and showing a loading spinner
- Then shows the placeholder of the new task list

![task-list-experiment](https://user-images.githubusercontent.com/2240960/152364068-dae2373b-5128-43e9-91ec-8e3d9a8d5fae.gif)

After writing these steps out I am thinking we might have to add some loading logic here: https://github.com/woocommerce/woocommerce-admin/blob/main/client/homescreen/layout.js#L167-L171
Or make sure that whatever loading icon takes up the entire screen afterwards doesn't.

It be nice if we could just show the spinner all this time until we show the new task list placeholder (if that makes sense?). Or really whatever doesn't cause any huge jumps on the screen.


There are a couple other issue's I noticed, that aren't specific to this PR, but are specific to the experiment:

- `class` used [here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/two-column-tasks/headers/appearance.js#L16) should be `className` I get a warning in my console.
- The extended task list shows up in every task (you can select either one of the paid **Product Types** or install **Google Listings & Ads** to have the extended task list show:
<img width="1345" alt="Screen Shot 2022-02-03 at 9 48 26 AM" src="https://user-images.githubusercontent.com/2240960/152355477-55af4f74-24a3-4469-a263-321c7fbba1e1.png">

ilyasfoo committed:
Fix class attribute typo

ilyasfoo committed:
Hide extended tasklist when task detail is shown

ilyasfoo committed:
Add local Suspense to stop lazy component bubbling up to Controller's Suspense

ilyasfoo committed:
Remove unused props

ilyasfoo committed:
Add missed Suspense, add loading logic to ActivityHeader

ilyasfoo commented:
Thank you for finding out the issues with this, @louwie17! I appreciate it!

>It be nice if we could just show the spinner all this time until we show the new task list placeholder (if that makes sense?). Or really whatever doesn't cause any huge jumps on the screen.

I found out the actual cause is that our lazy components were bubbling up to [Controller's Suspense](https://github.com/woocommerce/woocommerce-admin/blob/98a46ada0a04a73bf763aaa4971bb984d2b4e61c/client/layout/controller.js#L284), thus showing the app-wide Spinner component. I've fixed it in [25faab7](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/25faab7bd1103e51fb7d145ad357b3fce0319873) by adding a local Suspense to stop it from bubbling up. 

I've also fixed "Your store today" appearing in and out with [4762eb0](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/4762eb0848e37456073ca3731cdfacc15b83effa).

>class used [here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/two-column-tasks/headers/appearance.js#L16) should be className I get a warning in my console.

Addressed in [1db60db](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/1db60db2f8f27fb7b998ba8f1724414610dbdf17)

>The extended task list shows up in every task (you can select either one of the paid Product Types or install Google Listings & Ads to have the extended task list show:

Thanks for catching this! Fixed it in [2540f22](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/2540f2228b17e497f8a5e4f248f30b09ea3fccd6).

Please have another look!

louwie17 reviewed (approved):
Great work @ilyasfoo, thanks so much for the extra updates and fixes, things are looking pretty great now. Nice find on the `Suspense` issue, I wouldn't of guessed that!

Everything seems to test well for me now, so you got my approval. Feel free to merge it, it would be great though if someone else on @woocommerce/ghidorah could still do another round of testing with it on the beta 2 branch. But I am feeling pretty good about this now 💯 💯

ilyasfoo commented:
Thanks! Yes, we'll do another round of testing with this in beta 2.
